### PR TITLE
FP 0.21 Improve type checking and fix log_ingestion_delay

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Tested successfully with:
    mypy --ignore-missing-imports fastpath/core.py

### CODE REVIEW

**Comment by FedericoCeratto** [comment]
Deployed.

File: af/fastpath/fastpath/s3feeder.py [hunk 1]
```diff
@@ -154,11 +159,15 @@ def _cb(bytes_count):
         # TODO: handle missing file
         log.info("Downloading can %s size %d MB" % (fn, size / 1024 / 1024))
         diskf.parent.mkdir(parents=True, exist_ok=True)
+        tmpf = diskf.with_suffix(".s3tmp")
```
Comment (hellais) [inline af/fastpath/fastpath/s3feeder.py:162]: Do you think it makes sense to add something to clean these files up in the event of the process being restarted or killed?

Comment (FedericoCeratto) [reply af/fastpath/fastpath/s3feeder.py:162]: This is meant for manual runs and there is no automated cleanup of the cached files (given that the pipeline does not know if we want to use the cached files only once). At the moment I'm deleting the whole cache manually. In future we can add trimming old files and temp files.

**Review by hellais — APPROVED** [review]
LGTM

**Pull request merged**